src/core/multifeaturelistmodelbase.cpp

```diff
@@ -320,6 +319,15 @@ bool MultiFeatureListModelBase::canDeleteSelection()
   return !vlayer->readOnly() && ( vlayer->dataProvider()->capabilities() & QgsVectorDataProvider::DeleteFeatures ) && !vlayer->customProperty( QStringLiteral( "QFieldSync/is_geometry_locked" ), false ).toBool();
 }
 
+bool MultiFeatureListModelBase::canDuplicateSelection()
+{
+  if ( mSelectedFeatures.isEmpty() )
+    return false;
+
+  QgsVectorLayer *vlayer = mSelectedFeatures[0].first;
```

Comment:
Isn't it quite optimistic to think all the selected features are the same layer? Probably needs docs and ASSERT?

Reply:
Added a note, and fixed the issue above (whereas a duplicated feature would lead to an out-of-sync model filter / selected features' layer).